flink-java/src/test/java/org/apache/flink/api/java/utils/ParameterToolTest.java

```diff
@@ -0,0 +1,220 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.api.java.utils;
+
+import org.apache.flink.api.java.ClosureCleaner;
+import org.apache.flink.configuration.Configuration;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.Map;
+import java.util.Properties;
+
+public class ParameterToolTest {
+
+	@Rule
+	public TemporaryFolder tmp = new TemporaryFolder();
+
+
+	// ----- Parser tests -----------------
+
+	@Test(expected = RuntimeException.class)
+	public void testIllegalArgs() {
+		ParameterTool parameter = ParameterTool.fromArgs(new String[]{"berlin"});
+		Assert.assertEquals(0, parameter.getNumberOfParameters());
+	}
+
+	@Test
+	public void testNoVal() {
+		ParameterTool parameter = ParameterTool.fromArgs(new String[]{"-berlin"});
+		Assert.assertEquals(1, parameter.getNumberOfParameters());
+		Assert.assertTrue(parameter.has("berlin"));
+	}
+
+	@Test
+	public void testNoValDouble() {
+		ParameterTool parameter = ParameterTool.fromArgs(new String[]{"--berlin"});
+		Assert.assertEquals(1, parameter.getNumberOfParameters());
+		Assert.assertTrue(parameter.has("berlin"));
+	}
+
+	@Test
+	public void testMultipleNoVal() {
+		ParameterTool parameter = ParameterTool.fromArgs(new String[]{"--a", "--b", "--c", "--d", "--e", "--f"});
+		Assert.assertEquals(6, parameter.getNumberOfParameters());
+		Assert.assertTrue(parameter.has("a"));
+		Assert.assertTrue(parameter.has("b"));
+		Assert.assertTrue(parameter.has("c"));
+		Assert.assertTrue(parameter.has("d"));
+		Assert.assertTrue(parameter.has("e"));
+		Assert.assertTrue(parameter.has("f"));
+	}
+
+	@Test
+	public void testMultipleNoValMixed() {
+		ParameterTool parameter = ParameterTool.fromArgs(new String[]{"--a", "-b", "-c", "-d", "--e", "--f"});
+		Assert.assertEquals(6, parameter.getNumberOfParameters());
+		Assert.assertTrue(parameter.has("a"));
+		Assert.assertTrue(parameter.has("b"));
+		Assert.assertTrue(parameter.has("c"));
+		Assert.assertTrue(parameter.has("d"));
+		Assert.assertTrue(parameter.has("e"));
+		Assert.assertTrue(parameter.has("f"));
+	}
+
+	@Test(expected = IllegalArgumentException.class)
+	public void testEmptyVal() {
+		ParameterTool parameter = ParameterTool.fromArgs(new String[]{"--a", "-b", "--"});
+		Assert.assertEquals(2, parameter.getNumberOfParameters());
+		Assert.assertTrue(parameter.has("a"));
+		Assert.assertTrue(parameter.has("b"));
+	}
+
+	@Test(expected = IllegalArgumentException.class)
+	public void testEmptyValShort() {
+		ParameterTool parameter = ParameterTool.fromArgs(new String[]{"--a", "-b", "-"});
+		Assert.assertEquals(2, parameter.getNumberOfParameters());
+		Assert.assertTrue(parameter.has("a"));
+		Assert.assertTrue(parameter.has("b"));
+	}
+
+
+
+	/*@Test
```

Comment:
Do you want to keep this test as optional? If yes then it is better to have comments on why and when to uncomment this test. Otherwise, let's just remove it for now.